app/controllers/application_controller.rb

```diff
@@ -43,11 +43,11 @@ def use_javascript(js)
   end
 
   VALID_PAGES = ['last', 'unread']
-  def page
+  def page(allow_special: false)
     return @page if @page
     return (@page = 1) unless params[:page]
     @page = params[:page]
-    return @page if VALID_PAGES.include?(@page)
+    return @page if allow_special && VALID_PAGES.include?(@page)
```

Comment:
Oh that's clever

Reply:
Might be more succinct as a positional rather than argument parameter, looking at the calls later?

Reply:
Per Discord chat: seems clearer to read from the call location if we keep it as an argument parameter, gonna leave it be for now.